### PR TITLE
introducing .value_counts via .factorize; fixes #36

### LINKED ISSUE #36: add values_count support
naive version that uses factorize :

```python
def value_counts(self, normalize=False, sort=True, ascending=False, dropna=True):
    ind, u = self.factorize()
    missing = ind == -1
    
    if not dropna and np.any(missing):
        u = u.append(pd.Index(np.asarray([None])))
        ind[missing] = len(u) - 1
        cc = np.bincount(ind)
    else:
        cc = np.bincount(ind[np.logical_not(missing)])        

    if sort:
        ss_ = np.argsort(cc)
        if not ascending:
            ss_ = ss_[::-1]
    else:
        ss_ = slice(None, None)

    if normalize:
        cc = cc / cc.sum()

    return pd.Series(cc[ss_], index=u[ss_])
```
for general discussion, see https://github.com/pandas-dev/pandas/issues/22843

similar implems on : 
https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/categorical.py#L1464

### PR DESCRIPTION
**Pull request by artemru**

### CODE REVIEW

**Comment by artemru** [comment]
I need to add integration tests from pandas.

**Comment by artemru** [comment]
somebody has an idea for flake broken tests ?

File: fletcher/base.py [hunk 1]
```diff
@@ -872,6 +872,33 @@ def unique(self):
         """
         return type(self)(self.data.unique())
 
+    def value_counts(self, normalize=False, sort=True, ascending=False, dropna=True):
```
Comment (marc9595) [inline fletcher/base.py]: You need to add a docstring explaining what the function does to make flake happy
```suggestion
    def value_counts(self, normalize=False, sort=True, ascending=False, dropna=True):
    "Return a Series containing counts of unique values."
```

File: fletcher/base.py [hunk 2]
```diff
@@ -872,6 +872,34 @@ def unique(self):
...
+    "Return a Series containing counts of unique values."
```
Comment (krivonogov) [inline fletcher/base.py]:
```suggestion
        """Return a Series containing counts of unique values."""
```

File: fletcher/base.py [hunk 3]
```diff
@@ -872,6 +872,34 @@ def unique(self):
...
+        if not dropna and np.any(missing):
+            unique = type(self)(
+                pa.chunked_array(
+                    unique.data.chunks + [pa.array([None], type=self.dtype.arrow_dtype)]
+                )
+            )
+            ind[missing] = len(unique) - 1
+            cc = np.bincount(ind)
+        else:
+            cc = np.bincount(ind[np.logical_not(missing)])
```
Comment (krivonogov) [inline fletcher/base.py]: I would prefer more explicit names for `cc` and `ss_` variables

Comment (artemru) [reply fletcher/base.py]: done

**Comment by artemru** [comment]
any idea for broken ?

**Comment by krivonogov** [comment]
looks like test became longer than it was before. Value 200ms looks random, coming from https://hypothesis.readthedocs.io/en/latest/settings.html we can change it if needed I guess

**Pull request merged**